### PR TITLE
Limit Wiimote battery usage for Dolphin-lite users.

### PR DESCRIPTION
**Pull request by RachelBryk**
Wiimote batteries drain over time. The user must upgrade to Dolphin Pro for $19.95 to recharge them.

### CODE REVIEW

**Comment by SizzlingCalamari** [comment]
Devs should get a lifetime pro subscription I assume? :P

**Comment by johnwchadwick** [comment]
This is surely blocked on a lack of protection mechanisms. We need EXE packing and obfuscation as a part of our CI if we want to really make money.

**Comment by RamonSFe** [comment]
Hello Someone can Help Me? I Am Having Troubles With F-Zero GX for GameCube on Dolphin Emulator, Can Someone Help Me? I Can Pay in Real Money " Just if it is The Case " . I Am From Brazil .

**Comment by RachelBryk** [comment]
@RamonSFe please do not post here unless you have something productive to add, this is not a support forum.

**Comment by JosJuice** [comment]
@RamonSFe GitHub is _not_ the place to ask questions like these. Go to the forums instead: https://forums.dolphin-emu.org/

**Comment by RamonSFe** [comment]
@JosJuice Okay Thanks Again

**Comment by MoochMcGee** [comment]
lolwut I'd just recompile Dolphin with -DDOLPHINPRO as a compiler flag. Still, good joke.

**Comment by jimbo1qaz** [comment]
Blocking on Dolphin remaining open-source.

**Comment by delroth** [comment]
The cash must flow.

**Comment by Stevoisiak** [comment]
There's no code to deactivate the limits after April 1st.

All joking aside, if someone downloads this build, it will ask them to pay money, even if they hit the limit after April 1st is over, and they won't know to upgrade to a new build. To make matters worse, there isn't actually a way to get around the limit without manually editing the confit files.

I think that this should either be reverted as soon as possible, or a check should be put in to disable this after April 1st is over

File: Source/Core/DolphinWX/InputConfigDiag.cpp [hunk 1]
```diff
@@ -946,7 +946,14 @@ ControlGroupBox::ControlGroupBox(ControllerEmu::ControlGroup* const group, wxWin
 					setting->wxcontrol->Bind(wxEVT_SPINCTRL, &GamepadPage::AdjustSetting, eventsink);
 					options.push_back(setting);
 					wxBoxSizer* const szr = new wxBoxSizer(wxHORIZONTAL);
-					szr->Add(new wxStaticText(parent, wxID_ANY, wxGetTranslation(StrToWxStr(groupSetting->name))), 0, wxCENTER | wxRIGHT, 3);
+					wxStaticText* nameBox = new wxStaticText(parent, wxID_ANY, wxGetTranslation(StrToWxStr(groupSetting->name)));
+					if (groupSetting->name == "Battery" && !SConfig::GetInstance().m_DolphinPro)
+					{
+						setting->wxcontrol->Disable();
+						nameBox->SetToolTip(wxGetTranslation("Upgrade to Dolphin Pro� to recharge batteries."));
```
Comment (degasus) [inline Source/Core/DolphinWX/InputConfigDiag.cpp]: wtf:
../Source/Core/DolphinWX/InputConfigDiag.cpp:953:67: warning: illegal character encoding in string literal [-Winvalid-source-encoding]
                                                nameBox->SetToolTip(wxGetTranslation("Upgrade to Dolphin Pro<99> to recharge batteries."));

Seems like there is a broken character in this patch

Comment (RachelBryk) [inline Source/Core/DolphinWX/InputConfigDiag.cpp]: It is a ™, it shows up fine in dolphin, at least for me, on windows.

Comment (degasus) [inline Source/Core/DolphinWX/InputConfigDiag.cpp]: or as broken string in clang :/

Comment (lioncash) [inline Source/Core/DolphinWX/InputConfigDiag.cpp]: Also `wxGetTranslation("Upgrade to Dolphin Pro to recharge batteries.")` should be `_("Upgrade to Dolphin Pro to recharge batteries.")`

**Pull request merged**